corehq/apps/sms/api.py

```diff
@@ -397,6 +399,9 @@ def process_incoming(msg, delay=True):
         msg.couch_recipient_doc_type = v.owner_doc_type
         msg.couch_recipient = v.owner_id
         msg.domain = v.domain
+        user = CommCareUser.get(v.owner_id)
+        if 'location_id' in user:
+            add_msg_tags(msg, metadata=MessageMetadata(location_id=user.location_id, workflow=msg.workflow))
         msg.save()
```

Comment:
We also need to add the location_id here: https://github.com/lwyszomi/commcare-hq/blob/location_id_in_sms_logs/corehq/apps/sms/models.py#L73